apiserver/backrestservice/backrestimpl.go

```diff
@@ -469,6 +477,14 @@ func Restore(request *msgs.RestoreRequest, ns, pgouser string) msgs.RestoreRespo
 		return resp
 	}
 
+	///check if the current cluster is not upgraded to the deployed
+	//Operator version. If not, do not allow the command to complete
+	if cluster.Spec.UserLabels[config.LABEL_IS_UPGRADED] == msgs.False {
+		resp.Status.Code = msgs.Error
+		resp.Status.Msg = cluster.Name + msgs.UpgradeError
+		return resp
+	}
```

Comment:
I see this block of code enough, it seems like it should just be a general function, perhaps available in the common `apiserver` code:

```go
if err := checkClusterUpgradeStatus(cluster); err != nil {
  resp.Status.Code = msgs.Error
  resp.Status.Msg = err.Error()
  return resp
}
```

and the supporting function could be (and I may be screwing up the cluster type):

```go
func checkClusterUpgradeStatus(cluster crv1.Pgcluster) error {
    if cluster.Spec.UserLabels[config.LABEL_IS_UPGRADED] == msgs.False {
        return fmt.Errorf("%s %s", cluster.Name, UpgradeError)
    }

    return nil
}
```

with appropriate documentation, etc.

Reply:
@tjmoore4 FYI :+1: This is one that I think would help simplify some of the above changes as well, as you'd only have to make them in one place.